src/routes/accounts.ts

```diff
@@ -71,10 +71,20 @@ export default async function (req: Request, res: Response) {
 
   try {
     let result;
+    let paymentHash;
     try {
-      result = await submitPaymentWithTicket(payment, client, fundingWallet);
+      const signedPayment = await fundingWallet.sign(payment);
+      paymentHash = signedPayment.hash;
+      let response = await submitPaymentWithTicket(
+        payment,
+        client,
+        fundingWallet
+      );
+      ({ result, hash: paymentHash } = response);
```

Comment:
You can simplify this by ommitting the in-between variable since it's not used elsewhere.
```suggestion
      const { result, hash: paymentHash } = await submitPaymentWithTicket(
        payment,
        client,
        fundingWallet
      );
```

Reply:
can't do that because result is used outside of the try block

Reply:
I also would assume that the transactionHash would change after being attached to a ticket? @ckniffen - so the hash would change depends on which stage it errors out, before or after the ticket

Reply:
This result can't be used outside the try block as written - there's a separate response instantiated, but `let` has block scoping - so as soon as you're outside these brackets it stops existing.